lib/mrsk/cli/app.rb

```diff
@@ -50,12 +50,32 @@ def start
   desc "stop", "Stop app container on servers"
   def stop
     with_lock do
-      on(MRSK.hosts) do |host|
+      MRSK.hosts.each do |host|
+        roles = MRSK.roles_on(host)
+
+        roles.each do |role|
+          on(host) { execute *MRSK.auditor(role: role).record("Stopped app"), verbosity: :debug }
+          all_versions = list_versions(host: host, role: role)
+          all_versions.each do |version|
+            on(host) { execute *MRSK.app(role: role).stop(version: version), raise_on_non_zero_exit: false }
+          end
+        end
+      end
+    end
+  end
+
+  desc "stop_stale_containers", "Stop all app stale containers"
+  def stop_stale_containers
+    with_lock do
+      MRSK.hosts.each do |host|
         roles = MRSK.roles_on(host)
 
         roles.each do |role|
-          execute *MRSK.auditor(role: role).record("Stopped app"), verbosity: :debug
-          execute *MRSK.app(role: role).stop, raise_on_non_zero_exit: false
+          stale_versions = list_versions(host: host, role: role).drop(1)
+          stale_versions.each do |version|
+            say "Stopping stale container with version #{version}", :red
+            on(host) { execute *MRSK.app(role: role).stop(version: version), raise_on_non_zero_exit: false }
+          end
```

Comment:
Possible to share implementation with `stop`? Both are stopping containers, just with different versions.

Reply:
Good idea 🙏